### PR TITLE
Update documentation generation example in developer-notes.md

### PR DESCRIPTION
**Pull request by l0rinc**
Split out of https://github.com/bitcoin/bitcoin/pull/30454

In [the other readme](https://github.com/bitcoin/bitcoin/blob/6ce50fd9d0ae6850d54bf883e7a7c1bcb6912c5c/src/test/README.md?plain=1#L19) we've provided a default build directory instead.

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -338,9 +338,10 @@ Recommendations:
 
 ### Generating Documentation
 
-The documentation can be generated with `cmake --build <build_dir> --target docs`.
-The resulting files are located in `<build_dir>/doc/doxygen/html`; open
-`index.html` in that directory to view the homepage.
+Assuming the build directory is named `build`,
+the documentation can be generated with `cmake --build build --target docs`.
+The resulting files are located in `build/doc/doxygen/html`;
```
Comment (pablomartin4btc) [inline doc/developer-notes.md:343]: nit...
```suggestion
The resulting files will be located in `build/doc/doxygen/html`;
```

**Review by pablomartin4btc — COMMENTED** [review]
Concept ACK

It makes it consistent with the `src/test/README.md` file [mentioned](https://github.com/hebasto/bitcoin/pull/337#issue-2472099438) in the description.

**Comment by hebasto** [comment]
We'll proceed with this changes once https://github.com/bitcoin/bitcoin/pull/30454 is landed.

File: doc/developer-notes.md (hunk 1 above)
Comment (l0rinc) [reply doc/developer-notes.md:343]: Done, thanks

**Comment by hebasto** [comment]
Since https://github.com/bitcoin/bitcoin/pull/30454 has been merged, further development in this repository no longer makes sense. Please consider moving this PR to the main repository.

**Pull request closed without merging**